app/models/minion.rb

```diff
@@ -4,18 +4,43 @@
 
 # Minion represents the minions that have been registered in this application.
 class Minion < ApplicationRecord
+  class NotEnoughMinions < StandardError; end
+  class CouldNotAssignRole < StandardError; end
```

Comment:
Plus, it would be cool to add documentation on which cases we expect this exceptions to be raised.